core/src/components/textarea/textarea.ionic.scss

```diff
@@ -193,6 +197,21 @@ ion-icon {
   color: globals.$ion-primitives-neutral-500;
 }
 
+:host(.textarea-disabled.has-focus.ion-valid) {
+  --border-color: rgba(#{globals.$ion-semantics-success-base-rgb}, 0.6);
+}
+
+:host(.textarea-disabled.ion-touched.ion-invalid) {
+  --border-color: rgba(#{globals.$ion-semantics-danger-base-rgb}, 0.6);
+}
+
+:host(.textarea-disabled.has-focus.ion-valid) .textarea-bottom .helper-text,
+:host(.textarea-disabled.ion-touched.ion-invalid) .error-text,
+:host(.textarea-disabled.ion-color) .textarea-bottom .helper-text,
+:host(.textarea-disabled.ion-color) .helper-text {
+  opacity: 0.6;
```

Comment:
Instead of doing this, shouldn't the `.textarea-disabled `--color use `ion-state-disabled`?

Reply:
I ended up making a lot of changes for this because color wasn't working right for textarea or input:

https://github.com/ionic-team/ionic-framework/pull/30283/commits/96f0fd535f5e74bed88cf96a7ce53c6587f2bb99
https://github.com/ionic-team/ionic-framework/pull/30283/commits/f5fe36e608d2eaeac79f2b6376260874a5dea70a
https://github.com/ionic-team/ionic-framework/pull/30283/commits/76940df660fcdc65a220bc7abd85a1ee0b34b706

Updated snapshots!